AY2122S1-CS2103T-T17-3/tp
# Branch update delete command

Update delete command to delete by either Client Id or Email.
Updated UniquePersonsList to be able to remove Persons with multiple predicates as filters.

### Type of change


### How to test


Command | Behaviour
-- | --
delete i/2 | Should delete the client whose client id is 2
delete e/ben@gmail.com | Should delete the client whose email is ben@gmail.com
delete  i/2 e/ben@gmail.com | Should delete the client whose email is ben@gmail.com and client id is 2
delete i/{invalid clientid} | Should throw person not found exception
delete e/{invalid email} | Should throw person not found exception
delete i/2 e/{invalid email} | Should throw person not found exception even though person with client id equal to 2 exists
delete i/2 e/ben@gmail.com r/2 d/2000 | Should throw exception stating too many fields

### Checklist
